src/dotnet/README.md.j2

```diff
@@ -0,0 +1,176 @@
+
+- .NET CLI
+- .NET Runtime
+- ASP.NET Core Runtime
+
+Use this image for development, build and test if .NET and ASP.NET Core applications.
+{%- elif image.name == "dotnet-runtime" -%}
+This image contains the .NET runtimes and libraries and is optimized for running .NET applications in production.
+{%- elif image.name == "dotnet-aspnet" %}
+This image contains the ASP.NET Core and .NET runtimes and libraries,
+and is optimized for running ASP.NET Core appplications in production.
+{%- endif %}
+
+## Notice
+
+The .NET packages contained in this image come from a 3rd-party repository at
+[packages.microsoft.com](https://packages.microsoft.com).
+
+You can find the respective source code in [github.com/dotnet](https://github.com/dotnet).
+
+SUSE does not provide any support or warranties for this image.
+
+## Usage
+
+{% if image.name == "dotnet-sdk" -%}
+To compile an application, copy the sources and build the binary:
+
+```Dockerfile
+FROM {{ image.pretty_reference }} AS build
+WORKDIR /source
+
+# copy csproj and restore as distinct layers
+COPY *.csproj .
+RUN dotnet restore
+
+# copy and publish app and libraries
+COPY . .
+RUN dotnet publish --no-restore -o /app
+
+# final stage/image
+FROM registry.suse.com/bci/dotnet-runtime:{{ image.version }}
+WORKDIR /app
+COPY --from=build /app .
+
+# uncomment to run as non-root user
+# USER $APP_UID
+
+ENTRYPOINT ["./dotnetapp"]
+```
+
+Build and run the container image:
+
+```ShellSessionpodmanbuild -t my-dotnet-app .
+podman run -it --rm my-dotnet-app
+```
+
+There are situations, where you don't want to run an application inside a container.
+
+To compile the application, without running it inside a container instance, use the following command:
+
+```ShellSession
+podman run --rm -v "$PWD":/app:Z -w /app {{ image.pretty_reference }} dotnet run
+```
+
+To run unit tests in an isolated environment, use the following command:
+
+```ShellSession
+podman run --rm -v "$PWD":/app:Z -w /app/tests {{ image.pretty_reference }} dotnet test --logger:trx
+```
+{%- elif image.name == "dotnet-runtime" -%}
+To deploy an application, copy the sources and build the binary:
+
+```Dockerfile
+FROM registry.suse.com/bci/dotnet-sdk:{{ image.version }} AS build
+
+WORKDIR /source
+
+# copy csproj and restore as distinct layers
+COPY *.csproj .
+RUN dotnet restore
+
+# copy and publish app and libraries
+COPY . .
+RUN dotnet publish --no-restore -c Release -o /app
+
+# final stage/image
+FROM {{ image.pretty_reference }}
+
+WORKDIR /app
+COPY --from=build /app .
+
+# uncomment to run as non-root user
+# USER $APP_UID
+
+ENTRYPOINT ["./dotnetapp"]
+```
+
+Build and run the container image:
+
+```ShellSession
+podman build -t my-dotnet-app .
+podman run -it --rm my-dotnet-app
+```
+{%- elif image.name == "dotnet-aspnet" -%}
+To compile and deploy an application, copy the sources and build the binary:
+
+```Dockerfile
+FROM registry.suse.com/bci/dotnet-sdk:{{ image.version }} AS build
```

Comment:
can we avoid hardcoding the registry reference here?

Reply:
sure, what is the best approach?

Reply:
I changed to use the `registry` variable.